coalib/parsing/LineParser.py

```diff
@@ -62,8 +62,23 @@ def parse(self, line):
         # Escapes in value might be needed by the bears
         keys, value = self.__extract_keys_and_value(line)
 
+        # Add all the delimiters that stored as tuples
+        all_delimiters = self.key_value_delimiters
+        all_delimiters += self.key_delimiters
+        all_delimiters += self.comment_seperators
+        all_delimiters += self.section_override_delimiters
+        all_delimiters = "".join(all_delimiters)
+
+        # Add all keys and values in section_name_surroundings, which is
+        # stored as a dict
+        all_delimiters += "".join(self.section_name_surroundings.keys())
+        all_delimiters += "".join(self.section_name_surroundings.values())
```

Comment:
Added the keys and values from `section_name_surroundings`. Also added a comment for each group.